keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -160,7 +158,7 @@ Sidecar start up will follow these steps:
 1. Controller gets the `BucketClass` referenced by the `Bucket`.
 1. Controller creates a `BucketContent` object with its `BucketClassName` set to the name of its `BucketClass` and a `finalizer`.
 1. Sidecar detects the new `BucketContent` object and gets the associated `BucketClass`.
-1. Sidecar calls the GrantBucketAccess() rpc, passing the `bucketClass.objectBucketName` and the `bucketClass.parameters` and is returned a bucket endpoint and credentials.
+1. Sidecar calls the GrantBucketAccess() rpc, passing the `bucketClass.bucketIdentifier` and the `bucketClass.parameters` and is returned a bucket endpoint and credentials.
```

Comment:
should this be an interface to handle different identifiers for different object stores? Can we do that in an API?

Reply:
It's possible in go to marshal null interfaces, but the behavior is a little unpredictable (something that goes in as an int comes out as a float64 for example).  Couple that with the fact that we'd be passing it through a gRPC interface to something possibly written in another lang and I don't want to think about it.